ChemistryLib/PhreeqcKernel.cpp

```diff
@@ -22,15 +22,17 @@ namespace ChemistryLib
 {
 namespace PhreeqcKernelData
 {
-PhreeqcKernel::PhreeqcKernel(std::size_t const num_chemical_systems,
-                             std::vector<std::pair<int, std::string>> const&
-                                 process_id_to_component_name_map,
-                             std::string const database,
-                             AqueousSolution aqueous_solution,
-                             std::unique_ptr<Kinetics>
-                                 kinetic_reactants,
-                             std::vector<ReactionRate>&& reaction_rates)
+PhreeqcKernel::PhreeqcKernel(
+    std::size_t const num_chemical_systems,
+    std::vector<std::pair<int, std::string>> const&
+        process_id_to_component_name_map,
+    std::string const database,
+    AqueousSolution aqueous_solution,
+    std::unique_ptr<EquilibriumReactants>&& equilibrium_reactants,
```

Comment:
Not sure about r-value reference here. It is accessed in the function, but it is not moved. so it will be destroyed after the function finishes...

Reply:
Isn't for passing the ownership of the object?

Reply:
Yes, it is. `equilibrium_reactants` are now owned by the ctor. They are somehow used in the construction, which is fine.
But when the constructor ends, the unique_ptr will be destroyed. Is this intended behaviour?

Reply:
Thanks for the instruction.